pandas/io/pickle.py

```diff
@@ -163,18 +163,13 @@ def try_read(path, encoding=None):
                 # We want to silence any warnings about, e.g. moved modules.
                 warnings.simplefilter("ignore", Warning)
                 return read_wrapper(lambda f: pkl.load(f))
-        except Exception:
+        except Exception:  # noqa: E722
             # reg/patched pickle
-            try:
-                return read_wrapper(
-                    lambda f: pc.load(f, encoding=encoding, compat=False))
```

Comment:
is there a reason you elminated this branch here? IIRC this was a perf issue (IOW try the compat=False), but I wrote this code a long time ago.

Reply:
Yes, it appears that since then, someone has modified the load function so that compat isn't used at all.

https://github.com/pandas-dev/pandas/blob/5551bcf9d297ea8a0aeffb70b17ae6730e8abf89/pandas/compat/pickle_compat.py#L189-L214

Reply:
I am skeptical, I put this here because of performance issues. Can you run the asv for the pickle tests? though it might be only a case of reading py2 in py3 which is affected by this, which we prob don't perf check.

Reply:
can you also see if we actually hit this branch in any tests (meaning the try part is certainly hit, but the except is the question)